internal/cli/base.go

```diff
@@ -471,12 +471,7 @@ func remoteOpPreferred(ctx context.Context, client pb.WaypointClient, project *p
 	}
 	hasRemoteRunner := false
 	for _, runner := range runnersResp.Runners {
-		if !runner.Odr {
-			// NOTE(izaak): There is currently no way to distinguish between a remote runner and a CLI runner.
-			// So if some other waypoint client is performing an operation at this moment, we will interpret
-			// that as a remote runner, and this will return a false positive.
-
-			// Also note that this is designed to run before se start our own CLI runner.
+		if _, ok := runner.Kind.(*pb.Runner_Remote_); ok {
```

Comment:
I've always done `v, ok := ... ; ok && v != nil`. What you did is PROBABLY FINE, but talking out loud I've always wondered if this is totally fine haha. You can keep it how it is, just thinking out loud.

Reply:
oh ha interesting - i assume `v == nil` when `runner.Kind` has never been set - I think it's OK here because nil will still bypass this block.

Reply:
Well, you can do `runner.Kind = (*pb.Runner_Remote_)(nil)` to make `runner.Kind != nil` but once you cast it is `== nil`. I can’t think of any time this would happen without someone acting weird though. (This is the interface vs typed nil)